src/lib/RecordChunkIterator.spec.ts

```diff
@@ -134,6 +138,36 @@ describe('RecordChunkIterator', () => {
       jest.spyOn(iterator, 'afterEach').mockRejectedValue(err)
       await expect(iterator.process()).rejects.toThrow(err)
     })
+    test('emits error when callback takes too long', async () => {
+      jest.spyOn(RecordChunkIterator.prototype, 'emit')
+      jest.spyOn(RecordChunkIterator.prototype, 'beforeFirst').mockResolvedValueOnce(chunk)
+      jest.spyOn(console, 'warn').mockImplementationOnce((msg) => msg)
+      const callback: IteratorCallback = async (_, next) => {
+        await wait(11)
+        next()
+      }
+      const iterator = new RecordChunkIterator(session, callback, {
+        chunkSize: 10,
+        chunkTimeout: 10,
+      })
+      await iterator.process().catch((error) => {
+        expect(error).toBeInstanceOf(ChunkTimeoutError)
+      })
+      await wait()
```

Comment:
`ChunkTimeoutExpiredError` is emitted after the promise is rejected. So we need to wait here before running the assertion.